tests/prague/eip2537_bls_12_381_precompiles/test_bls12_g1mul.py

```diff
@@ -0,0 +1,240 @@
+            id="invalid_point_1",
+        ),
+        pytest.param(
+            PointG1(Spec.P1.x, Spec.P1.y - 1) + Scalar(0),
+            id="invalid_point_2",
+        ),
+        pytest.param(
+            PointG1(Spec.P1.x, Spec.P1.y + 1) + Scalar(0),
+            id="invalid_point_3",
+        ),
+        pytest.param(
+            PointG1(Spec.P1.x, Spec.P1.x) + Scalar(0),
+            id="invalid_point_4",
+        ),
+        pytest.param(
+            b"\x80" + bytes(Spec.INF_G1)[1:] + Scalar(0),
+            id="invalid_encoding",
+        ),
+        pytest.param(
+            (Spec.INF_G1 + Scalar(0))[:-1],
+            id="input_too_short",
+        ),
+        pytest.param(
+            b"\x00" + (Spec.INF_G1 + Scalar(0)),
+            id="input_too_long",
+        ),
+        pytest.param(
+            b"",
+            id="zero_length_input",
+        ),
+        pytest.param(
+            Spec.P1_NOT_IN_SUBGROUP + Scalar(1),
+            id="bls_g1mul_not_in_subgroup_1",
+        ),
+        pytest.param(
+            Spec.P1_NOT_IN_SUBGROUP_TIMES_2 + Scalar(1),
+            id="bls_g1mul_not_in_subgroup_2",
+        ),
+        pytest.param(
+            Spec.P1_NOT_IN_SUBGROUP_TIMES_2 + Scalar(Spec.Q),
+            id="bls_g1mul_not_in_subgroup_times_q",
+        ),
+        pytest.param(
+            Spec.P1_NOT_IN_SUBGROUP + Scalar(Spec.Q),
+            id="bls_g1mul_not_in_subgroup_times_q_2",
+        ),
+        pytest.param(
+            Spec.G1 + Spec.G1,
+            id="bls_g1_add_input_invalid_length",
+        ),
+        pytest.param(
+            Spec.G2 + Spec.G2,
+            id="bls_g2_add_input_invalid_length",
+        ),
+        pytest.param(
+            Spec.G1,
+            id="bls_g1_truncated_input",
+        ),
+    ],
+)
+@pytest.mark.parametrize("expected_output", [Spec.INVALID], ids=[""])
+def test_invalid(
+    state_test: StateTestFiller,
+    pre: dict,
+    post: dict,
+    tx: Transaction,
+):
+    """
+    Negative tests for the BLS12_G1MUL precompile.
+    """
+    state_test(
+        env=Environment(),
+        pre=pre,
+        tx=tx,
+        post=post,
+    )
+
+
+@pytest.mark.parametrize(
+    "input,expected_output,precompile_gas_modifier",
+    [
+        pytest.param(
+            Spec.INF_G1 + Scalar(0),
+            Spec.INF_G1,
+            1,
+            id="extra_gas",
+        ),
+        pytest.param(
+            Spec.INF_G1 + Scalar(0),
+            Spec.INVALID,
+            -1,
+            id="insufficient_gas",
+        ),
+    ],
+)
+def test_gas(
+    state_test: StateTestFiller,
+    pre: dict,
+    post: dict,
+    tx: Transaction,
+):
+    """
+    Test the BLS12_G1MUL precompile gas requirements.
+    """
+    state_test(
+        env=Environment(),
+        pre=pre,
+        tx=tx,
+        post=post,
+    )
+
+
+@pytest.mark.parametrize(
+    "call_opcode",
+    [
+        Op.STATICCALL,
+        Op.DELEGATECALL,
+        Op.CALLCODE,
```

Comment:
For the call types tests, should we use `Op.CALL` too? Maybe I am missing something regarding the `call_opcode` fixture though

Reply:
It's the default type of call for the rest of the tests, so I thought that it would be redundant to use it in this array too.